### DISCUSSION TITLE
Allow editing relational value through readonly relational interface

### DISCUSSION THREAD

**Discussion by cah4a** [directus/directus#8401] [Feature Requests]
Navigation to the relation detailed form is working only when the field is not marked as read-only.

In many cases, there is a need to make a relation interface read-only while allowing editing of related data.
For example, `order.user_id` should be read-only. But admin still wants to have fast navigation to the related user detailed form.

That affects all relational interface types.

### To Reproduce

1. Create table `users`
2. Create table `orders`
3. Add `user_id` field in `orders` table. Make it read-only.
4. Open order detailed form
5. Click on the user card

Actual result: Nothing happens
Expected result: User information form is opened

**Comment by rijkvanzanten** [comment]
Any edits that are made in the relational item are staged/saved as a nested object change under the relational field in order to track revisions.

**Reply by cah4a** [reply]
Why not a plain redirect to `/collections/${collection}/${id}` then?

**Reply by joselcvarela** [reply]
Hey @cah4a 
While it seems to be easy to implement, we should discuss the possible scenarios.
For example, you are suggesting that if a relation is readonly, the items in the list should be clickable and on click it should redirect to their page.
But it can also be, open the drawer and you can edit the item in the drawer (the relation is readonly but item could be not)
Or even, open the drawer and show a info view about the item (like activity records).

**Reply by cah4a** [reply]
Yep, any of the options will work in my case.
The best I could think of is making it configurable, so you could choose what fits best for a particular relation.

**Reply by benhaynes** [reply]
Generally, it's better to open items in a drawer to avoid leaving the parent edit page (which forces saving or abandoning changes). We can also add a "launch" action button to all drawers to actually go to the item _page_, but that should be a secondary option.

**Reply by cah4a** [reply]
That would be awesome!

**Reply by rijkvanzanten** [reply]
To extend on @benhaynes answer, another crucially important part is permissions. You might have edit access to a particular data point through a parent relation, but not to the related collection directly (keeping https://github.com/directus/directus/discussions/3144 in mind)

**Comment by w0kyj** [comment]
Read-Only use case for Data Sharing is impacted by this click restriction also.
Using the example above add another table products
Update orders to include a O2M to products

Now if I want to Data Share the order, but click to see related product details, it is not currently possible.

**Reply by hugopvl** [reply]
The same problem happens to me, I need to share information with related items, but it is impossible to see related items.
Any suggestion ?

But in my case I needed to be able to select which related items I wanted to share, and I think I could solve this issue if the share id or share name could be filtered, in the rules, is that possible?

Thanks for listening.

**Reply by joselcvarela** [reply]
Any idea @Oreilles ? 🤔

**Reply by rijkvanzanten** [reply]
@joselcvarela This should be a small tweak in how the o2m/m2m interfaces work. Right now, their disabled states also prevents the click to open the drawer. Instead, we could simply still render the drawer, but make that readonly as well. That way, you can still read and explore all related data, without editing it 👍🏻

**Reply by hugopvl** [reply]
> @joselcvarela This should be a small tweak in how the o2m/m2m interfaces work. Right now, their disabled states also prevents the click to open the drawer. Instead, we could simply still render the drawer, but make that readonly as well. That way, you can still read and explore all related data, without editing it 👍🏻

@rijkvanzanten Did you always make this change? Or is it still under review?
Thanks for listening

**Reply by rijkvanzanten** [reply]
I haven't personally had the time to implement this tweak. We'll gladly accept a PR, or it'll be triaged and implemented according to our normal priorities 👍🏻

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
